### PR TITLE
Fix camera reset bug

### PR DESCRIPTION
**Pull request by Sunketchupm**
Fix the camera reset issue as described in issue #494. This not only fixes that particular freecam issue, but also the camera reset issues when popping from a bubble in general.

### CODE REVIEW

File: src/game/bettercamera.inc.h [hunk 1]
```diff
@@ -122,26 +122,28 @@ s16 newcam_saved_defmode = -1;
 extern bool gDjuiInMainMenu;
 
 ///This is called at every level initialisation.
-void newcam_init(struct Camera *c, UNUSED u8 dv) {
+void newcam_init(struct Camera *c, u8 softResetted) {
```
Comment (Isaac0-dev) [inline src/game/bettercamera.inc.h]: "resetted" is not a word

File: src/game/camera.c [hunk 2]
```diff
@@ -3627,8 +3631,12 @@ void init_camera(struct Camera *c) {
     c->yaw = gLakituState.yaw;
     c->nextYaw = gLakituState.yaw;
 
-    newcam_init(c, 0);
+    newcam_init(c, sSoftResettingCamera);
     newcam_init_settings();
+
+    if (sSoftResettingCamera) {
+        sSoftResettingCamera = FALSE;
```
Comment (Isaac0-dev) [inline src/game/camera.c]: no condition necessary, just `sSoftResettingCamera = FALSE;` will be fine

File: src/game/camera.c [hunk 3]
```diff
@@ -12337,6 +12345,22 @@ void mode_rom_hack_camera(struct Camera *c) {
     pos[1] = mPos[1] + desiredHeight;
     pos[2] = mPos[2] + sins(sRomHackYaw) * desiredDist;
 
+    // Move camera down for hangable ceilings
+    if (sMarioCamState->action & ACT_FLAG_HANGING) {
```
Comment (Isaac0-dev) [inline src/game/camera.c:12346]: not sure what happened here, but isn't this code already merged?

File: src/game/camera.c [hunk 4]
```diff
@@ -3449,6 +3449,7 @@ void reset_camera(struct Camera *c) {
 
     if (gFirstPersonCamera.enabled) {
         gFirstPersonCamera.yaw = gMarioStates[0].faceAngle[1] + 0x8000;
+        set_mario_action(&gMarioStates[0], ACT_IDLE, 0);
```
Comment (Isaac0-dev) [inline src/game/camera.c]: what are you trying to fix by doing this?

File: src/game/camera.c (hunk 3 above)
Comment (Sunketchupm) [reply src/game/camera.c:12346]: Not sure either

File: src/game/camera.c (hunk 4 above)
Comment (Sunketchupm) [reply src/game/camera.c]: I forgot I made this change, but it is supposed to fix Mario getting stuck if the camera resets while in first person

**Pull request merged**